pkg/controller/worker/machineset_vmo.go

```diff
@@ -147,7 +147,7 @@ func (w *workerDelegate) cleanupVmoDependencies(ctx context.Context, infrastruct
 
 func cleanupOrphanVMODependencies(ctx context.Context, client azureclient.Vmss, dependencies []azureapi.VmoDependency, resourceGroupName string) error {
 	vmoListAll, err := client.List(ctx, resourceGroupName)
-	if err != nil {
+	if err != nil && !azureclient.IsAzureAPINotFoundError(err) {
```

Comment:
Is it possible to add a unit test to make sure that we won't regress this in the future?